stl/inc/memory

```diff
@@ -1504,6 +1550,233 @@ private:
     }
 };
 
+#ifdef __cpp_concepts
+template <class _Ty>
+constexpr _Ty _Div_ceil(_Ty _Val1, _Ty _Val2) noexcept {
+    using _Signed   = make_signed_t<_Ty>;
+    const auto _Res = _STD div(_Signed(_Val1), _Signed(_Val2));
+    return (_Res.rem == 0) ? (_Res.quot) : (_Res.quot + 1);
+}
+
+template <class _NoThrowIt>
+struct _Uninitialized_rev_destroying_backout { // struct to undo partially constructed ranges in _Uninitialized_xxx
+                                               // algorithms
+    _NoThrowIt _First;
+    _NoThrowIt _Last;
+
+    explicit _Uninitialized_rev_destroying_backout(_NoThrowIt _Dest) : _First(_Dest), _Last(_Dest) {}
+
+    _Uninitialized_rev_destroying_backout(_NoThrowIt _First_, _NoThrowIt _Last_) : _First(_First_), _Last(_Last_) {}
+
+    _Uninitialized_rev_destroying_backout(const _Uninitialized_rev_destroying_backout&) = delete;
+    _Uninitialized_rev_destroying_backout& operator=(const _Uninitialized_rev_destroying_backout&) = delete;
+
+    ~_Uninitialized_rev_destroying_backout() {
+        while (_Last != _First) {
+            --_Last;
+            _STD destroy_at(_Last);
+        }
+    }
+
+    template <class... _Types>
+    void _Emplace_back(_Types&&... _Vals) { // construct a new element at *_Last and increment
+        _Construct_in_place(*_Last, _STD forward<_Types>(_Vals)...);
+        ++_Last;
+    }
+
+    _NoThrowIt _Release() { // suppress any exception handling backout and return _Last
+        _First = _Last;
+        return _Last;
+    }
+};
+
+template <class _Ty>
+void _Reverse_destroy_multidimensional_n(_Ty* const _Arr, size_t _Size) {
+    while (_Size > 0) {
+        --_Size;
+        // TRANSITION, LLVM-42404
+        if constexpr ((rank_v<_Ty>) > 0) {
+            _Reverse_destroy_multidimensional_n(_Arr[_Size], extent_v<_Ty>);
+        } else {
+            _Destroy_in_place(_Arr[_Size]);
+        }
+    }
+}
+
+template <class _Ty, size_t _Size>
+void _Uninitialized_copy_multidimensional(const _Ty (&_In)[_Size], _Ty (&_Out)[_Size]) {
+    // TRANSITION, LLVM-42404
+    if constexpr ((rank_v<_Ty>) > 0) {
+        for (size_t _Idx = 0; _Idx < _Size; ++_Idx) {
+            _TRY_BEGIN
+            _Uninitialized_copy_multidimensional(_In[_Idx], _Out[_Idx]);
+            _CATCH_ALL
+            _Reverse_destroy_multidimensional_n(_Out, _Idx);
+            _RERAISE;
+            _CATCH_END
+        }
+    } else if constexpr (_Ptr_copy_cat<_Ty*, _Ty*>::_Really_trivial) {
+        if (0 < _Size) {

```

Comment:
That is a strang way of writing, wspecially if all the while loops have it switched
```suggestion
        if (_Size > 0) {
```
Appears below too

Reply:
I think the (0 < _Size) form in this PR is the canonical one in this codebase. Examples:

https://github.com/microsoft/STL/blob/master/stl/inc/algorithm#L103
https://github.com/microsoft/STL/blob/master/stl/inc/algorithm#L656
https://github.com/microsoft/STL/blob/master/stl/inc/algorithm#L1646
https://github.com/microsoft/STL/blob/master/stl/inc/xmemory#L1721

Reply:
I vaguely recall that we've removed many Yoda

Reply:
I guess I don't view it as yoda conditionals as much as preferring `<` and `<=` wherever possible.

Reply:
Yes, that's the "operators point left" style I finally got around to mentioning at the end of my rambling comment.

Reply:
Done